environment.yml

```diff
@@ -9,59 +9,73 @@ dependencies:
   - python-dateutil>=2.5.0
   - pytz
 
-  # development
+  # benchmarks
   - asv
+
+  # building
   - cython>=0.28.2
+
+  # code checks
   - flake8
-  - flake8-comprehensions
-  - flake8-rst>=0.6.0,<=0.7.0
-  - gitpython
-  - hypothesis>=3.82
-  - ipywidgets
-  - isort
-  - moto
+  - flake8-comprehensions  # used by flake8, linting of unnecessary comprehensions
+  - flake8-rst>=0.6.0,<=0.7.0  # linting of code blocks in rst files
+  - isort  # check that imports are in the right order
   - mypy
+  - pycodestyle  # used by flake8
+  - pip:
+    - cpplint
```

Comment:
This seems to be available through conda-forge, if we want to switch to that: https://anaconda.org/conda-forge/cpplint. Understood if you want to keep this a purely a reorganization though.